File: src/main/kotlin/com/back/domain/chat/chat/entity/RoomParticipant.java
package com.back.domain.chat.chat.entity;

import com.back.domain.member.entity.Member;
import com.back.global.jpa.entity.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@NoArgsConstructor
@Setter
public class RoomParticipant extends BaseEntity {
    // room , user 메니투원 설정
    @ManyToOne
    private ChatRoom chatRoom;

    @ManyToOne
    private Member member;

    private LocalDateTime leftAt;

    private boolean isActive;

    public RoomParticipant(ChatRoom chatRoom, Member member) {
        this.chatRoom = chatRoom;
        this.member = member;
        this.isActive = true;
    }

    public ChatRoom getChatRoom() {
        return chatRoom;
    }

    public Member getMember() {
        return member;
    }

    public LocalDateTime getLeftAt() {
        return leftAt;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setChatRoom(ChatRoom chatRoom) {
        this.chatRoom = chatRoom;
    }

    public void setMember(Member member) {
        this.member = member;
    }

    public void setLeftAt(LocalDateTime leftAt) {
        this.leftAt = leftAt;
    }

    public void setActive(boolean active) {
        isActive = active;
    }
}

File: src/main/kotlin/com/back/domain/chat/chat/entity/RoomParticipant.kt
package com.back.domain.chat.chat.entity

import com.back.domain.member.entity.Member
import com.back.global.jpa.entity.BaseEntity
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
class RoomParticipant(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chat_room_id")
    val chatRoom: ChatRoom,
    
    @ManyToOne(fetch = FetchType.LAZY) 
    @JoinColumn(name = "member_id")
    val member: Member
) : BaseEntity() {
    
    var leftAt: LocalDateTime? = null
        private set
    
    var isActive: Boolean = true
        private set
    
    // 채팅방 나가기
    fun leave() {
        this.leftAt = LocalDateTime.now()
        this.isActive = false
    }
    
    // 채팅방 재참여
    fun rejoin() {
        this.leftAt = null
        this.isActive = true
    }
    
    // 활성화 (기존 채팅방 재사용시)
    fun activate() {
        this.isActive = true
        this.leftAt = null
    }
}
